R/aes.R

```diff
@@ -130,14 +129,9 @@ new_aesthetic <- function(x, env = globalenv()) {
 
   x
 }
-new_aes <- function(x, env = globalenv()) {
-  check_object(x, is.list, "a {.cls list}")
-  x <- lapply(x, new_aesthetic, env = env)
-  structure(x, class = "uneval")
-}
 
 #' @export
-print.uneval <- function(x, ...) {
+`print.ggplot2::mapping` <- function(x, ...) {
```

Comment:
When we have an S3 generic we need for both S3 and S7 objects, we need to implement methods for the S7 objects in the S3 fashion, because using `S7::method()` will invalidate the S3 methods.

Reply:
It's been tagged with the S7 v0.3.0 milestone here: https://github.com/RConsortium/S7/issues/390. I don't know anything about the specifics of the timeline though.